src/main/java/io/FileReader.java

```diff
@@ -0,0 +1,7 @@
+package io;
+
+import java.util.List;
+
+public interface FileReader {
```

Comment:
I see you have two interfaces within IO package. Why did you put the implementations within service/impl ?